### PR TITLE
FIX: Disable 'Create Topic' button if tag is staff-only.

### PR DESCRIPTION
**Pull request by nbianca**

### CODE REVIEW

File: app/serializers/tag_serializer.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 class TagSerializer < ApplicationSerializer
-  attributes :id, :name, :topic_count
+  attributes :id, :name, :topic_count, :staff
+
+  def staff
+    DiscourseTagging.staff_tag_names.include?(name)
```
Comment (ZogStriP) [inline app/serializers/tag_serializer.rb]: Did you check this isn't creating an N+1?

Comment (SamSaffron) [reply app/serializers/tag_serializer.rb]: yeah this looks pretty risky to me:

https://github.com/discourse/discourse/blob/74d2d4f658171bcab41ccd7257495e7e39fb45e4/lib/discourse_tagging.rb#L196-L202

**Comment by SamSaffron** [comment]
got to sort out the N+1 here prior to merging also double check where we are using this serializer, we want to make sure it is not being used on standard topic list pages and so on.

File: app/assets/javascripts/discourse/routes/tags-show.js.es6 [hunk 2]
```diff
@@ -111,13 +111,15 @@ export default Discourse.Route.extend({
     ).then(list => {
       if (list.topic_list.tags) {
         tag.set("id", list.topic_list.tags[0].name); // Update name of tag (case might be different)
+        tag.set("staff", list.topic_list.tags[0].staff);
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/routes/tags-show.js.es6]: `setProperties` here instead?

**Comment by SamSaffron** [comment]
This looks good to me, let's try it out!

**Pull request merged**